src/checks.c

```diff
@@ -1586,6 +1586,11 @@ static int connect_conn_chk(struct task *t)
 			if (s->proxy->options2 & PR_O2_CHK_SNDST)
 				b_putblk(&check->bo, trash.area,
 					 httpchk_build_status_header(s, trash.area, trash.size));
+			/* set the host header per backend */
+			if (s->vhost)
```

Comment:
braces around clause implied by indentation ?

Reply:
Yes, just committed a patch.
Great catch. Thank you.